### PR TITLE
BRE-40 Add action for reporting upcoming release version to Slack

### PR DESCRIPTION
**Pull request by urbinaalex17**
## 🎟️ Tracking

Jira: https://bitwarden.atlassian.net/browse/BRE-40 
Notify #release with version for upcoming release

## 🚧 Type of change

-   🚀 New feature development

## 📔 Objective

To help enable better communication and visibility, and to align the entire company expectations, we would like to send a slack notification to #release with the projects and their corresponding version bumps.

## 📋 Code changes

-   Add **report-upcoming-release-version/action.yml:** Initial GitHub Action based on the [report-deployment-status-to-slack action](https://github.com/bitwarden/gh-actions/blob/main/report-deployment-status-to-slack/action.yml)

## 📸 Screenshots

image

https://bitwarden.slack.com/archives/C05T0840415/p1715884688154529

## ⏰ Reminders before review

-   Contributor guidelines followed
-   All formatters and local linters executed and passed
-   Written new unit and / or integration tests where applicable
-   Used internationalization (i18n) for all UI strings
-   CI builds passed
-   Communicated to DevOps any deployment requirements
-   Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

-   👍 (`:+1:`) or similar for great changes
-   📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
-   ❓ (`:question:`) for questions
-   🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
-   🎨 (`:art:`) for suggestions / improvements
-   ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
-   🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
-   ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: report-upcoming-release-version/action.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  slack-channel:
+    description: "The name of the Slack channel."
+    default: release
+  AZURE_KV_CI_SERVICE_PRINCIPAL:
+    description: "The service principal used to authenticate to Azure."
+    required: true
+outputs:
+  channel_id:
+    description: "Channel ID where the Slack message was posted"
+    value: ${{ steps.slack-message.outputs.channel_id }}
```
Comment (mimartin12) [inline report-upcoming-release-version/action.yml:19]: ⛏️  Grammar change
```suggestion
    description: "Channel ID where the Slack message was posted."
    value: ${{ steps.slack-message.outputs.channel_id }}
```

File: report-upcoming-release-version/action.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+runs:
+  using: "composite"
+  steps:
+    - name: Login to Azure - Prod Subscription
```
Comment (mimartin12) [inline report-upcoming-release-version/action.yml:24]: ❓ Can we always assume the type of subscription here?
`name: Login to Azure - Prod Subscription`

Maybe something like this wouldn't assume details that may not be correct.
```suggestion
    - name: Login to Azure
```

File: report-upcoming-release-version/action.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          local version="${{ inputs.version }}"
+          local project="${{ inputs.project }}"
+          SLACK_MESSAGE=":meow_attention: Upcoming release \`$version\` for $project."
+        }
+
+        slack_message_fn
+
+        echo 'slack_message<<EOF' >> $GITHUB_OUTPUT
+        echo "$SLACK_MESSAGE" >> $GITHUB_OUTPUT
+        echo 'EOF' >> $GITHUB_OUTPUT
```
Comment (mimartin12) [inline report-upcoming-release-version/action.yml:50]: ⛏️  Consistency recommendations.
```suggestion
        echo "slack_message<<EOF" >> $GITHUB_OUTPUT
        echo "$SLACK_MESSAGE" >> $GITHUB_OUTPUT
        echo "EOF" >> $GITHUB_OUTPUT
```

**Review by mimartin12 — CHANGES_REQUESTED** [review]
Various changes around consistency and grammar.

I would also like to see some tests for this, similar to how we are doing it with `report-deployment-status-to-slack` action. Can you include those in this PR so that we can have tests runs on this prior to changes?

**Review by mimartin12 — CHANGES_REQUESTED** [review]
The changes look good, thanks @alex8bitw!

However, we are still missing tests. A simple add would be similar to this. https://github.com/bitwarden/gh-actions/blob/main/.github/workflows/test-report-deployment-status-to-slack.yml. These should run when you make changes to the composite action, can you please add them to this PR?

**Comment by urbinaalex17** [comment]
> The changes look good, thanks @urbinaalex17 !
> 
> However, we are still missing tests. A simple add would be similar to this. https://github.com/bitwarden/gh-actions/blob/main/.github/workflows/test-report-deployment-status-to-slack.yml. These should run when you make changes to the composite action, can you please add them to this PR?

This PR is ready to be merged! Thanks for the suggestions, @mimartin12!

**Pull request merged**